src/discussions/data/hooks.js

```diff
@@ -50,7 +50,7 @@ export const useSidebarVisible = () => {
     return true;
   }
 
-  if (isViewingTopics || isViewingLearners) {
+  if ((isViewingTopics || isViewingLearners) && totalThreads > 0) {
```

Comment:
@AhtishamShahid This solution is not an optimal solution. This change is breaking things in locations. I checkout your branch on my local machine and this is happing. Please check the video. 
https://www.loom.com/share/27abf0da19424bd38c8daef5673b4a39